_plugins/related_posts.rb

```diff
@@ -0,0 +1,62 @@
+# https://github.com/jumanji27/related_posts-jekyll_plugin
+
+require 'jekyll/post'
+
+module Jekyll
+  module RelatedPostsByTags
+    # Used to remove #related_posts so that it can be overridden
+    def self.included(klass)
+      klass.class_eval do
+        remove_method :related_posts
+      end
+    end
+
+    # Calculate related posts.
+    # Returns [<Post>]
+    def related_posts(posts)
+      return [] unless posts.size > 1
+      highest_freq = tag_freq(posts).values.max
+      related_scores = Hash.new(0)
+
+      posts.each do |post|
+        post.tags.each do |tag|
+          if self.tags.include?(tag) && post != self
+            cat_freq = tag_freq(posts)[tag]
+            related_scores[post] += (1+highest_freq-cat_freq)
+          end
+        end
+      end
+
+      sort_related_posts(related_scores)
+    end
+
+    # Calculate the frequency of each tag.
+    # Returns {tag => freq, tag => freq, ...}
+    def tag_freq(posts)
+      return @tag_freq if @tag_freq
+      @tag_freq = Hash.new(0)
+      posts.each do |post|
+        post.tags.each {|tag| @tag_freq[tag] += 1}
+      end
+      @tag_freq
+    end
+
+    # Sort the related posts in order of their score and date
+    # and return just the posts
+    def sort_related_posts(related_scores)
+      related_scores.sort do |a,b|
+        if a[1] < b[1]
+          1
+        elsif a[1] > b[1]
+          -1
+        else
+          b[0].date <=> a[0].date
+        end
+      end.collect {|post,freq| post}
```

Comment:
Unused block argument - `freq`. If it's necessary, use `_` or `_freq` as an argument name to indicate that it won't be used.<br>Prefer `map` over `collect`.<br>Avoid multi-line chains of blocks.<br>Space missing after comma.<br>Space between { and | missing.<br>Space missing inside }.